common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java

```diff
@@ -0,0 +1,179 @@
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.chat.MessageType;
+import com.wynntils.core.chat.RecipientType;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.core.notifications.NotificationManager;
+import com.wynntils.wc.event.ChatMessageReceivedEvent;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@FeatureInfo
+public class InfoMessageFilterFeature extends UserFeature {
+    private static final Pattern PRE_WELCOME_1 = Pattern.compile("^§7Loading Resource Pack...$");
+    private static final Pattern PRE_WELCOME_2 =
+            Pattern.compile("^§6Thank you for using the WynnPack. Enjoy the game!$");
+    private static final Pattern PRE_WELCOME_3 = Pattern.compile(
+            "^§cSelect a class! Each class is saved individually across all servers, you can come back at any time with /class and select another class!$");
+
+    private static final Pattern WELCOME_1 = Pattern.compile("^ +§6§lWelcome to Wynncraft!$");
+    private static final Pattern WELCOME_2 = Pattern.compile("^ +§fplay.wynncraft.com §7-/-§f wynncraft.com$");
+
+    private static final Pattern SYSTEM_INFO = Pattern.compile("^(§r)?§.\\[Info\\] .*$");
+
+    private static final Pattern SOUL_POINT_1 = Pattern.compile("^§5As the sun rises, you feel a little bit safer...$");
+    private static final Pattern SOUL_POINT_2 = Pattern.compile("^§d\\[(\\+\\d+ Soul Points?)\\]$");
+
+    private static final Pattern LEVEL_UP_1 = Pattern.compile("^§6.* is now level .* in §.*$");
+    private static final Pattern LEVEL_UP_2 =
+            Pattern.compile("^§8\\[§r§7!§r§8\\] §r§7Congratulations to §r.* for reaching (combat )?§r§flevel .*!$");
+
+    private static final Pattern VIP_LOGIN =
+            Pattern.compile("^§.\\[§r§.[A-Z+]+§r§.\\] §r§..*§r§. has just logged in!$");
+
+    private static final Pattern BACKGROUND_WELCOME_1 = Pattern.compile("^ +§6§lWelcome to Wynncraft!$");
+    private static final Pattern BACKGROUND_WELCOME_2 =
+            Pattern.compile("^ +§fplay.wynncraft.com §7-/-§f wynncraft.com$");
+
+    private static final Pattern BACKGROUND_SYSTEM_INFO = Pattern.compile("^(§r§8)?\\[Info\\] .*$");
+
+    private static final Pattern BACKGROUND_SOUL_POINT_1 =
+            Pattern.compile("^(§r§8)?As the sun rises, you feel a little bit safer...$");
+    private static final Pattern BACKGROUND_SOUL_POINT_2 = Pattern.compile("^§r§7\\[(\\+\\d+ Soul Points?)\\]$");
+
+    private static final Pattern BACKGROUND_LEVEL_UP_1 = Pattern.compile("^(§r§8)?.* is now level .* in §.*$");
+    private static final Pattern BACKGROUND_LEVEL_UP_2 =
+            Pattern.compile("^(§r§8)?\\[!\\] Congratulations to §r.* for reaching (combat )?§r§7level .*!$");
+
+    private static final Pattern BACKGROUND_VIP_LOGIN =
```

Comment:
I assume this works for every rank? Perhaps rename this?

Reply:
Yes, agree.